app/scripts/services/authorization.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("AuthorizationService", function($q, Logger, $interval, DataService, APIService){
+    function AuthorizationService() {
+      this.rules = null;
+      this.intervalTimer = null;
+      this.forceReload = false;
+      this.canICreateSomeResource = false;
+      this.projectNameRules = null;
+    }
+
+    var normalizeRules = function(rules) {
```

Comment:
https://github.com/openshift/origin/blob/master/pkg/authorization/api/v1/types.go#L18 is your best bet. 

verbs can contain specific verbs (like `create`) and/or `*`
resources can contain specific resources (like `pods`) and/or `*`
apiGroups can contain specific groups (like "" for the v1 origin/k8s groups) or `*`
resourceNames can contain specific names (like `mypod`) or `*` or be empty/missing (which means the same as `*`)

For your purposes (99% correct estimates about whether you can do something, erring on the side of letting you try and fail) you can probably ignore attributeRestrictions and nonResourceURLs